src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,93 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double MAX_CAPACITY = 0.75;
+    private static final int INDEX_OF_CAPACITY = 2;
+    private int size;
+    private Node<K, V>[] elements;
+
+    {
+        elements = (Node<K, V>[]) new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        resizeIfNeed();
+        int currentCapacity = elements.length;
+
+        int hash = hashKey(key);
+
+        Node<K, V> currentNode = elements[hash];
```

Comment:
The hash index calculation should use `elements.length` to ensure the index is within the bounds of the array. Consider using `Math.abs(hashKey(key) % elements.length)` to calculate the correct index.